codec/src/main/java/io/netty/handler/codec/AsciiString.java

```diff
@@ -656,7 +501,7 @@ public boolean equalsIgnoreCase(CharSequence string) {
         }
 
         for (int i = 0; i < thisLen; i++) {
-            char c1 = (char) (value[i] & 0xFF);
+            char c1 = (char) value[i];
```

Comment:
Yes you are right...a bit overzealous with the mask removal :)  Seemed strange to mask an 8 bit value with an 8 bit mask...forgot about sign extension.  These loops with conversions are not ideal as the following conversion is done on every index `byte -> sign extended int -> char`....